packages/config/src/load.js

```diff
@@ -88,10 +88,13 @@ export async function loadNuxtConfig ({
   }
 
   // Load Combine configs
-  // Priority: configOverrides > nuxtConfig > .nuxtrc > .nuxtrc (global)
+  // Priority: configOverrides > nuxtConfig > .nuxt/dist/.nuxtrc > .nuxtrc > .nuxtrc (global)
+  const dev = configOverrides.dev ?? options.dev ?? configContext.dev
+  const buildDir = configOverrides.buildDir ?? options.buildDir ?? configContext.buildDir ?? '.nuxt'
   options = defu(
     configOverrides,
     options,
+    !dev ? rc.read({ name: '.nuxtrc', dir: path.resolve(buildDir, 'dist') }) : {},
```

Comment:
One edge case of using dotfiles in `dist` is previously (during the time was using [name] in assets) we had lots of reports that CI/CD pipeline is ignoring them for deployment. It might be probably safer to use `nuxtrc` (not `.nuxtrc`)